### PR TITLE
Implement support import `fs` promises from root of fs module

### PR DESCRIPTION
**Pull request by floydspace**
To improve compatibility with many packages using fs promises, I implemented the compatibility with the following way of import
```ts
import {promises as fs} from 'fs';
```
it does not have any breaking changes, just refactored a bit the fs module
What do you think?

### CODE REVIEW

**Comment by richarddavison** [comment]
This is a step in the right direction!
Callbackify existing fs/promises should also be possible. This would be simplest to do from js.

**Comment by floydspace** [comment]
> ... This would be simplest to do from js.

yes, the `llrt` runtime consumer can workaround it on js side, however, it will lead to making a lot of patches for the third-party libraries, or making runtime-specific implementation, like it is usually done with `bun` or `deno`.
But if we could make `llrt` as compatible with `node` as possible, the enabling `llrt` usage could be much effortless.

**Comment by richarddavison** [comment]
Sorry I was not clear. I mean it should definitely be added to LLRT just that it can be implemented in JS rather than rust (callbackify)

File: build.mjs [hunk 1]
```diff
@@ -36,7 +36,7 @@ const ES_BUILD_OPTIONS = {
     "uuid",
     "hex",
     "os",
-    "fs/promises",
```
Comment (richarddavison) [inline build.mjs:39]: Don't we need to specify both `fs/promises` and `fs`? as they are separate modules

**Review by richarddavison — CHANGES_REQUESTED** [review]

File: build.mjs (hunk 1 above)
Comment (floydspace) [reply build.mjs:39]: `fs/promises` will be excluded by default as the child path of `fs`.
Trust me, it was my contribution to `esbuild`.
https://github.com/evanw/esbuild/pull/186

**Review by richarddavison — APPROVED** [review]

**Pull request merged**